src/components/KCodeBlock/KCodeBlock.vue

```diff
@@ -0,0 +1,829 @@
+        :appearance="regExpButtonAppearance"
+        :aria-pressed="isRegExpMode"
+        :is-rounded="false"
+        size="small"
+        title="Use regular expression (Alt+R)"
+        data-testid="k-regexp-mode-button"
+        @click="toggleRegExpMode"
+      >
+        <span class="k-visually-hidden">RegExp mode enabled</span>
+
+        .*
+      </KButton>
+
+      <KButton
+        class="k-filter-mode-button"
+        type="button"
+        icon="filter"
+        :appearance="isFilterMode ? 'secondary' : 'outline'"
+        :aria-pressed="isFilterMode"
+        :is-rounded="false"
+        size="small"
+        title="Filter results (Alt+F)"
+        data-testid="k-filter-mode-button"
+        @click="toggleFilterMode"
+      >
+        <template #icon>
+          <KIcon
+            class="k-button-icon"
+            icon="filter"
+            size="16"
+            title="Filter results (Alt+F)"
+            color="currentColor"
+          />
+        </template>
+
+        <span class="k-visually-hidden">Filter mode enabled</span>
+      </KButton>
+
+      <KButton
+        class="k-previous-match-button"
+        type="button"
+        :is-rounded="false"
+        size="small"
+        title="Previous match (Shift+Enter)"
+        :disabled="matchingLineNumbers.length === 0 || isFilterMode"
+        data-testid="k-previous-match-button"
+        @click="jumpToPreviousMatch"
+      >
+        <template #icon>
+          <KIcon
+            class="k-button-icon"
+            icon="chevronUp"
+            size="16"
+            title="Previous match (Shift+Enter)"
+            color="currentColor"
+          />
+        </template>
+
+        <span class="k-visually-hidden">Previous match</span>
+      </KButton>
+
+      <KButton
+        class="k-next-match-button"
+        type="button"
+        :is-rounded="false"
+        size="small"
+        title="Next match (Enter)"
+        :disabled="matchingLineNumbers.length === 0 || isFilterMode"
+        data-testid="k-next-match-button"
+        @click="jumpToNextMatch"
+      >
+        <template #icon>
+          <KIcon
+            class="k-button-icon"
+            icon="chevronDown"
+            size="16"
+            title="Previous match (Shift+Enter)"
+            color="currentColor"
+          />
+        </template>
+
+        <span class="k-visually-hidden">Next match</span>
+      </KButton>
+    </div>
+
+    <div class="k-code-block__content">
+      <!-- eslint-disable vue/no-v-html -->
+      <pre
+        v-if="isShowingFilteredCode"
+        class="k-filtered-code-block"
+        data-testid="k-filtered-code-block"
+      ><span
+        v-if="props.showLineNumbers"
+        class="k-line-number-rows"
+      >
+        <a
+          v-for="line in matchingLineNumbers"
+          :id="`${linePrefix}-L${line}`"
+          :key="line"
+          :href="`#${linePrefix}-L${line}`"
+        >{{ line }}</a>
+      </span><code v-html="filteredCode" /></pre>
```

Comment:
I'm not sure of a way around this, but I don't love outputting `html` here, XSS vulnerability. Same comment for other instances of `v-html`

Reply:
The way around this would be introducing a sanitizing library and setting `codeElement.innerHTML` with its result. This _could_ be viable because DOM-based libraries like PrismJS wouldn’t conflict with it: Prism itself will set `code.innerHTML` so any already set HTML from `<code v-html>` would be overwritten. And this can be generalized: This component expects raw code in `props.code` and not HTML content that should be rendered. This could change should this component take up syntax highlighting responsibilities but that’s another story.

I’ll have a look what the cost of adding such a library would be.

Reply:
Makes sense; I'm good with this.